src/prefect/engine/task_runner.py

```diff
@@ -901,28 +901,28 @@ def get_task_run_state(
         except signals.LOOP as exc:
             new_state = exc.state
             assert isinstance(new_state, Looped)
-            new_state.result = Result(
-                value=new_state.result, result_handler=self.result_handler
+            new_state.result = Result( # todo: here
+                value=new_state.result, result_handler=self.result_handler # todo: here
             )
             new_state.cached_inputs = inputs
             new_state.message = exc.state.message or "Task is looping ({})".format(
                 new_state.loop_count
             )
             return new_state
 
-        result = Result(value=result, result_handler=self.result_handler)
+        result = Result(value=result, result_handler=self.result_handler) # todo: here
         state = Success(
-            result=result, message="Task run succeeded.", cached_inputs=inputs
+            result=result, message="Task run succeeded.", cached_inputs=inputs # todo: here
         )
 
         ## checkpoint tasks if a result_handler is present, except for when the user has opted out by disabling checkpointing
         if (
             state.is_successful()
             and prefect.context.get("checkpointing") is True
             and self.task.checkpoint is not False
-            and self.result_handler is not None
+            and self.result_handler is not None # todo: here
         ):
-            state._result.store_safe_value()
+            state._result.store_safe_value() # todo: here
```

Comment:
This is also a write call; now that we are presumably greedily writing results anyway, this might not be necessary 🤔